website/common/script/ops/buyGear.js

```diff
@@ -35,6 +35,13 @@ module.exports = function buyGear (user, req = {}, analytics) {
     throw new NotAuthorized(i18n.t('equipmentAlreadyOwned', req.language));
   }
 
+  let gearNumber = Number(item.key.slice(-1));
```

Comment:
Why are you using in the first place .splice with a fixed number of chars selected and in the second a regexp that can match a different number of charachters?

Isn't the regexp going to cause problems if the key includes other numbers? Since all keys follow the same form (`type_class_number`) can you take the gearNumber from `key.split('_')[2]`, covert it to a number and check that it's not Nan as there are also gear items without a number part like `eyewear_special_summerWarrior`

Reply:
And use the same method to generate the previous gear key